src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -149,37 +149,55 @@ void Run (DirectoryAssemblyResolver res)
 			var java    = new Dictionary<string, TypeDefinition> ();
 			var acw_map = new StreamWriter (temp_map_file);
 
+			var managedConflicts = new Dictionary<string, List<string>> ();
+			var javaConflicts    = new Dictionary<string, List<string>> ();
+
 			foreach (var type in java_types) {
 				string managedKey = type.FullName.Replace ('/', '.');
 				string javaKey    = JavaNativeTypeManager.ToJniName (type).Replace ('/', '.');
 
 				acw_map.WriteLine ("{0};{1}", type.GetPartialAssemblyQualifiedName (), javaKey);
 
 				TypeDefinition conflict;
+				bool hasConflict = false;
 				if (managed.TryGetValue (managedKey, out conflict)) {
-					Log.LogWarning (
-							"Duplicate managed type found! Mappings between managed types and Java types must be unique. " +
-							"First Type: '{0}'; Second Type: '{1}'.",
-							conflict.GetAssemblyQualifiedName (),
-							type.GetAssemblyQualifiedName ());
-					Log.LogWarning (
-							"References to the type '{0}' will refer to '{1}'.",
-							managedKey, conflict.GetAssemblyQualifiedName ());
-					continue;
+					if (!managedConflicts.TryGetValue (managedKey, out var list))
+						managedConflicts.Add (managedKey, list = new List<string> { conflict.GetPartialAssemblyName () });
+					list.Add (type.GetPartialAssemblyName ());
+					hasConflict = true;
 				}
 				if (java.TryGetValue (javaKey, out conflict)) {
-					Log.LogError (
-							"Duplicate Java type found! Mappings between managed types and Java types must be unique. " +
-							"First Type: '{0}'; Second Type: '{1}'",
-							conflict.GetAssemblyQualifiedName (),
-							type.GetAssemblyQualifiedName ());
+					if (!javaConflicts.TryGetValue (javaKey, out var list))
+						javaConflicts.Add (javaKey, list = new List<string> { conflict.GetAssemblyQualifiedName ()});
+					list.Add (type.GetAssemblyQualifiedName ());
 					keep_going = false;
-					continue;
+					hasConflict = true;
+				}
+				if (!hasConflict) {
+					managed.Add (managedKey, type);
+					java.Add (javaKey, type);
+					acw_map.WriteLine ("{0};{1}", managedKey, javaKey);
+					acw_map.WriteLine ("{0};{1}", JavaNativeTypeManager.ToCompatJniName (type).Replace ('/', '.'), javaKey);
 				}
-				managed.Add (managedKey, type);
-				java.Add (javaKey, type);
-				acw_map.WriteLine ("{0};{1}", managedKey, javaKey);
-				acw_map.WriteLine ("{0};{1}", JavaNativeTypeManager.ToCompatJniName (type).Replace ('/', '.'), javaKey);
+			}
+
+			foreach (var kvp in managedConflicts) {
+				Log.LogCodedWarning (
+					"XA4214",
+					"The managed type `{0}` exists in more than one assembly: {1}. " +
+					"Please refactor the managed type names in these assemblies so that they are not identical.",
```

Comment:
The current message mentions what will be used:

    References to the type `Foo.Bar` will refer to `Foo.Bar, Baz`.

This information is lost by this PR.  It should be preserved.

This might be a scenario where we want multiple warnings for the "same" issue:

    warning XA4214: The managed type `Foo.Bar` exists in more than one assembly: Baz, Qux, Quux.
    warning XA4214: References to the type `Foo.Bar` will refer to `Foo.Bar, Baz`.